build-scripts/config_common.cmake

```diff
@@ -597,3 +597,6 @@ endif()
 if (NOT WAMR_BUILD_SANITIZER STREQUAL "")
   message ("     Sanitizer ${WAMR_BUILD_SANITIZER} enabled")
 endif ()
+if (ENABLE_NEON EQUAL 1)
+  add_definitions(-DENABLE_NEON=1)
+endif()
```

Comment:
Most of the devices we target are ARMv7 rather than ARMv8 (and even the v8 devices don't usually run in AArch64 mode. Most, but not all of the v7 devices have NEON support.

So I want to add a new macro for this, but not sure if this is the right thing to do?

I was thinking I could do something like a `WAMR_BUILD_ARM_NEON` style option and then `#ifdef WAMR_BUILD_ARM_NEON != 0 || BUILD_TARGET_AARCH64` or something like that?

Reply:
I think the new build flag makes sense if really NEON support is non-deterministic only based on the architecture. I'd however suggest a few changes here:
* use `WAMR_BUILD_*` prefix for consistency (e.g. `WAMR_BUILD_ARM_NEON`)
* consider enabling it by default for some common combinations
* for the `#ifdef WAMR_BUILD_ARM_NEON != 0 || BUILD_TARGET_AARCH64` condition, I'm not sure if the build target is needed; we should just use the NEON ifdef, otherwise it should not be defined if not desired.